homeassistant/components/snmp/switch.py

```diff
@@ -162,6 +162,15 @@ async def async_setup_platform(
     request_args = await async_create_request_cmd_args(
         hass, auth_data, UdpTransportTarget((host, port)), baseoid
     )
+    # Switches are read-only if command OID is not set which means they can't be turned on or off
+    # however since CONF_COMMAND_OID is marked as optional, we need to check if it is set
+    command_args: RequestArgsType | None
+    if command_oid:
+        command_args = await async_create_request_cmd_args(
+            hass, auth_data, UdpTransportTarget((host, port)), command_oid
+        )
+    else:
+        command_args = None
```

Comment:
This config makes no sense but it would be a breaking change to make `CONF_COMMAND_OID` required